libraries/Wire/src/Wire.h

```diff
@@ -81,12 +83,45 @@ class TwoWire: public Stream
     static void onReceiveService(uint8_t, uint8_t*, size_t, bool, void *);
 #endif /* SOC_I2C_SUPPORT_SLAVE */
     bool initPins(int sdaPin, int sclPin);
-    bool allocateWireBuffer(void);
-    void freeWireBuffer(void);
+    bool allocateWireBuffer();
+    void freeWireBuffer();
 
 public:
     TwoWire(uint8_t bus_num);
     ~TwoWire();
+
+    void begin() override final
+    {
+        begin(-1, -1);
+    }
+
+	/**
+	 * Fixme: it is necessary to determine whether it is possible to use without SOC_I2C_SUPPORT_SLAVE? !!!
+	 */
+    void begin(uint8_t address) override final
+    {
+#if SOC_I2C_SUPPORT_SLAVE
+        begin(address, -1, -1, 0);
+#else
+        log_e("I2C slave is not supported by you SoC!!!");
+#endif
+    }
```

Comment:
This override in specific is only used with SOC_I2C_SUPPORT_SLAVE. For master mode the address is set in `beginTransmission`.

Reply:
ok. 
Then it is necessary to change to :
```
#if SOC_I2C_SUPPORT_SLAVE
        begin(address, -1, -1, 0);
#else
        beginTransmission(address);
#endif
```
???

Reply:
You can use `#if SOC_I2C_SUPPORT_SLAVE` around the entire function implementation and declaration. That way it won't be compiled if it does not have support for I2C slave mode.

Reply:
You can put the check in the abstract class as well.

Reply:
> You can put the check in the abstract class as well.

Then it will not be compatible with the Arduino protocol.

Reply:
I thought that the whole point of this was to implement the Arduino interface

Reply:
I still don't understand what you are trying to do. Your PR description only mentions fixing the requestFrom methods. What is your objective with all these changes in the PR ? Could you please update the description with your reasoning ?

Reply:
https://github.com/espressif/arduino-esp32/pull/8818#issuecomment-1829914025

Reply:
Okay, thanks for the clarification. You don't need to change anything. The way you did is correct. Just remove the fixme comment

Reply:
So, just for me to confirm, you are changing everything to match the arduino API, correct ?

Reply:
> So, just for me to confirm, you are changing everything to match the arduino API, correct ?

yes.